controllers/front/payment.php

```diff
@@ -330,19 +330,42 @@ private function getBits($blockonomics, $crypto, $currency, $total)
         return $bits;
     }
 
-    private function displayError($blockonomics)
+    private function displayError($blockonomics, $responseObj=NULL)
```

Comment:
This function can be simplified to avoid repetition and complete the checks in a single statement
```
$unable_to_generate = '<h3>' . $blockonomics->l(
    'Could not generate new address',
    'payment'
) . '</h3><p>';

if (isset($responseObj)
    && isset($responseObj->data)
    && isset($responseObj->data->message)
    && strpos(strtolower($responseObj->data->message), 'gap limit') !== false 
    || strpos(strtolower($responseObj->data->message), 'temporary') !== false) {
    $unable_to_generate .= $responseObj->data->message;
}else{
    $unable_to_generate .= $blockonomics->l(
        'Please use Test Setup button in configuration to diagnose the error ',
        'payment'
    );
}

echo $unable_to_generate;
die();
```

Reply:
I'll check and update on how feasible it is to do it in frontend. As far as I saw before the code's exiting before reaching the frontend, so maybe adding it might add unnecessary complexity.